src/d3a/models/strategy/infinite_bus.py

```diff
@@ -45,33 +44,41 @@ def __init__(self, energy_sell_rate=None, energy_rate_profile=None, energy_buy_r
 
     def _set_market_maker_rate(self):
         if self.energy_rate_profile is not None:
-            GlobalConfig.market_maker_rate = \
-                read_and_convert_identity_profile_to_float(self.energy_rate_profile)
-        elif isinstance(self.energy_rate, (int, float)):
+            GlobalConfig.market_maker_rate = read_arbitrary_profile(
+                InputProfileTypes.IDENTITY, self.energy_rate_profile)
+        elif self.energy_rate is not None:
             GlobalConfig.market_maker_rate = self.energy_rate
-        elif isinstance(self.energy_rate, (str, dict)):
-            GlobalConfig.market_maker_rate = \
-                read_arbitrary_profile(InputProfileTypes.IDENTITY, self.energy_rate)
         else:
-            GlobalConfig.market_maker_rate = \
-                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE
+            GlobalConfig.market_maker_rate = (
+                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE)
```

Comment:
I know this is not new code, and I only see it now.
Imagine a user sets the `GlobalConfig.market_maker_rate` in a setup file, you would overwrite it by adding an infinite bus and don't provide an `energy_sell_rate`. Is this wanted? 
If you agree that it is not please remove this block. Thanks!

Reply:
As far as I know `InfiniteBus` is meant to supersede for setting the `GlobalConfig.market_maker_rate`. Have you already seem a setup where user didn'T set `energy_sell_rate ` but goes the other way via setting `GlobalConfig.market_maker_rate` ?

Reply:
I haven't, I just thought about this corner case. And yes, the `InfiniteBus is meant to supersedes` the setting of the `GlobalConfig.market_maker_rate` even with my proposal. But if you "defuse" it and don't provide a energy_rate it should not overwrite the global config that the user set before.